src/ldapclient_lib.py

```diff
@@ -0,0 +1,395 @@
+# Copyright 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Library for the ldap-client relation.
+
+This library contains the Provides and Requires classes for handling the ldap-client interface.
+
+
+### Provider Charm
+
+### Requirer Charm
+
+
+"""
+
+import logging
+import pathlib
+import shlex
+import subprocess
+import zipfile
+
+from ops.charm import (
+    CharmBase,
+    CharmEvents,
+    RelationBrokenEvent,
+    RelationChangedEvent,
+    RelationJoinedEvent,
+)
+from ops.framework import EventBase, EventSource, Handle, Object
+from ops.model import ActiveStatus, MaintenanceStatus, ModelError
+
+logger = logging.getLogger(__name__)
+
+"""
+Events - TBD
+"""
+
+# class LdapClientProviderCharmEvents(CharmEvents):
+#    """Events the LDAP Client provider can leverage."""
+# No custom events necessary for the provider, just has to pass along
+# information to the client.
+
+
+class GlauthSnapReadyEvent(EventBase):
+    """Charm Event triggered when a glauth snap is ready to start."""
+
+    def __init__(
+        self,
+        handle: Handle,
+    ):
+        super().__init__(handle)
```

Comment:
I think having `__init__` here is unnecessary since all you are doing is just invoking the parent constructor for event base:

```suggestion
class GlauthSnapReadyEvent(EventBase):
    """Charm Event triggered when a glauth snap is ready to start."""
```